src/core_plugins/tile_map/public/maps_visualization.js

```diff
@@ -272,10 +272,40 @@ export function MapsVisualizationProvider(serviceSettings, Notifier, getAppState
 
       const indexPatternName = agg.vis.indexPattern.id;
       const field = agg.fieldName();
-      const filter = { meta: { negate: false, index: indexPatternName } };
-      filter[filterName] = { ignore_unmapped: true };
-      filter[filterName][field] = filterData;
-      getAppState().$newFilters = [filter];
+      const query = this.vis.API.queryManager.getQuery();
+      const language = query.language;
+
+      if (language === 'lucene') {
+        const filter = { meta: { negate: false, index: indexPatternName } };
+        filter[filterName] = { ignore_unmapped: true };
+        filter[filterName][field] = filterData;
+
+        this.vis.API.queryFilter.addFilters([filter]);
+      }
+      else if (language === 'kuery') {
+        const { fromKueryExpression, toKueryExpression, nodeTypes } = this.vis.API.kuery;
```

Comment:
wow, this seems super complicated in comparison to the lucene filters

Reply:
That's because kuery gives the vis full control over query manipulations instead of relying on bad assumptions and leaky abstractions :)